components/core/src/package/ident_target.rs

```diff
@@ -0,0 +1,82 @@
+use std::{cmp::{Ord,
+                Ordering,
+                PartialOrd},
+          fmt,
+          ops::Deref,
+          str::FromStr};
+
+use super::{PackageIdent,
+            PackageTarget};
+use crate::error::{Error,
+                   Result};
+
+// A package ident doesn't uniquely identify an artifact; we need the target as well.
+//
+#[derive(Debug, Clone, Eq, Hash, PartialEq)] // Copy needs PackageIdent::Copy
+pub struct PackageIdentTarget {
+    // Ideally these would not be pub, but the accessors seem to hit issues with split borrows
+    pub ident:  PackageIdent,
+    pub target: PackageTarget,
+}
+
+impl PackageIdentTarget {
+    pub fn new(ident: PackageIdent, target: PackageTarget) -> Self {
+        PackageIdentTarget { ident, target }
+    }
+
+    // These would be cool, except that they don't work
+    //  println!("Fetching {} for {}", p.ident(), p.target()); hits
+    //  E0308 or other borrow related errors, probably because they
+    //  are opaque and the checker uses the lifetime of the whole
+    //  struct and doesn't treat them as disjoint entities.
+
+    //  https://doc.rust-lang.org/nomicon/borrow-splitting.html
+    //
+    //    pub fn ident(&self) -> PackageIdent { &self.ident }
+    //    pub fn target(&self) -> PackageTarget { &self.target }
+
+    /// Generates the name of the hab package (hart) file.
+    pub fn archive_name(&self) -> Result<String> {
+        self.ident.archive_name_with_target(self.target)
+    }
+}
+
+// It would be nice if Ident and Target implemented Ord, PartialOrd and PartialEq
+impl Ord for PackageIdentTarget {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let ord = self.ident.by_parts_cmp(&other.ident);
+        match ord {
+            Ordering::Equal => self.target.deref().cmp(&other.target.deref()),
+            _ => ord,
+        }
+    }
+}
+
+impl PartialOrd for PackageIdentTarget {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> { Some(self.cmp(other)) }
+}
+
+impl fmt::Display for PackageIdentTarget {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}/{}", self.ident, self.target)
+    }
+}
+
+// PackageIdentTargets are ORIGIN/NAME/VERSION/DATE/TARGET
```

Comment:
We should get a design doc and input from the community as well as there are significant decisions.  Might be better to take this out of this PR and do something separately.

Reply:
Removed PackageIdentTarget from PR